dm/pkg/utils/common.go

```diff
@@ -289,6 +293,9 @@ func UnpackTableID(id string) *filter.Table {
 	}
 }
 
+var _ sessionctx.Context = (*session)(nil)
+var _ sqlexec.RestrictedSQLExecutor = (*session)(nil)
```

Comment:
what does those for?

Reply:
our `session` type will be cast to those interfaces in TiDB's function, so checking cast here is more friendly than panic

Reply:
it won't pass static checking if it doesn't impl those interface, or are we passing by `interface{}` and there's force cast somewhere?

Reply:
> it won't pass static checking if it doesn't impl those interface

This. https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance

Reply:
> This. https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance

that's useful when the `type` never used as that `interface` such as a library, else compiler would complain

Reply:
it doesn't matter here